### PR TITLE
add client class tested

### PR DESCRIPTION
**Pull request by 914-Mereu-Bianca**
- [x] add google style comments

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -2,11 +2,20 @@
...
+    suite/suite.cpp
+    client_test.cpp
+    mainwindow_test.cpp
 )
+
+set(PROJECT_HEADERS
+    client_test.h
+    mainwindow_test.h)
+
+qt6_wrap_cpp(MOC_SOURCES ${PROJECT_HEADERS})
```
Comment (CristianIonutIonescu) [inline test/CMakeLists.txt:15]: shouldn't it be MOCK?

File: test/client_test.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#include "client_test.h"
+
+class RTopServerImpl final : public data::RTopServer::Service {
+    grpc::Status StreamData(grpc::ServerContext* context, const data::RequestTopPackage* request, grpc::ServerWriter<data::TopPackage>* writer) override {
+        data::TopPackage response;
+        data::GeneralInfo gi;
+        *response.mutable_general_info() = gi;
```
Comment (CristianIonutIonescu) [inline test/client_test.cpp:7]: why dereference when you can use
response->mutable_general_info()?

**Review by CristianIonutIonescu — COMMENTED** [review]
minor issues to be addressed

File: test/CMakeLists.txt (hunk 1 above)
Comment (914-Mereu-Bianca) [reply test/CMakeLists.txt:15]: I understood that the files generated are for the Meta-Object Compiler (moc), i don't think is the same as MOCK

Comment (CristianIonutIonescu) [reply test/CMakeLists.txt:15]: My bad, thought you were thinking about mock files

File: test/client_test.cpp (hunk 2 above)
Comment (914-Mereu-Bianca) [reply test/client_test.cpp:7]: when using '->' operator i get the error "member reference type 'data::TopPackage' is not a pointer", but by dereferencing it works

**Review by CristianIonutIonescu — APPROVED** [review]

**Pull request merged**